static/js/publisher/metrics/graphs/activeDevices.js

```diff
@@ -1,201 +1,223 @@
-/* globals bb, moment */
-
-import { formatAxis, formatXAxisTickLabels, formatYAxisTickLabels } from '../axis';
-import debounce from '../../../libs/debounce';
-import { snapcraftGraphTooltip, positionTooltip } from '../tooltips';
-import { PADDING, COLOR_SCALE } from '../config';
+import { select as d3Select, mouse as d3Mouse } from 'd3-selection';
+import { format as d3Format } from 'd3-format';
+import { utcParse as d3TimeParse, utcFormat as d3TimeFormat } from 'd3-time-format';
+import { scaleLinear as d3ScaleLinear, scaleOrdinal as d3ScaleOrdinal } from 'd3-scale';
+import { schemePaired as d3SchemePaired } from 'd3-scale-chromatic';
+import { area as d3Area, stack as d3Stack, curveMonotoneX as d3CurveMonotoneX, stackOrderReverse as d3StackOrderReverse } from 'd3-shape';
+import { extent as d3Extent, bisector as d3Bisector } from 'd3-array';
+import { axisBottom as d3AxisBottom, axisLeft as d3AxisLeft } from 'd3-axis';
+import { cullXAxis, cullYAxis } from './helpers';
 
 function showGraph(el) {
-  formatAxis(el);
   el.style.opacity = 1;
 }
 
-export default function activeDevices(days, activeDevices, type) {
-  const el = document.getElementById('active_devices');
+export default function activeDevices(activeDevices) {
+  // Basic svg setup
+  const holder = document.querySelector('svg').parentNode;
```

Comment:
It seems like we are assuming here that there is only one `svg` in the document. This may break if something is added, even in completely unrelated part of the page. Would be good to have more specific component related selector.

Reply:
Thanks, this was a bad copy/paste that I did change, but then pasted over again!